common/djangoapps/util/module_utils.py

```diff
@@ -13,17 +13,21 @@ def yield_dynamic_descriptor_descendents(descriptor, module_creator):  # pylint:
 
     while len(stack) > 0:
         next_descriptor = stack.pop()
-        stack.extend(get_dynamic_descriptor_children(next_descriptor, module_creator))
+        stack.extend(get_dynamic_descriptor_children(next_descriptor, user_id, module_creator))
         yield next_descriptor
 
 
-def get_dynamic_descriptor_children(descriptor, module_creator, usage_key_filter=None):
+def get_dynamic_descriptor_children(descriptor, user_id, module_creator=None, usage_key_filter=None):
     """
     Returns the children of the given descriptor, while supporting descriptors with dynamic children.
     """
     module_children = []
     if descriptor.has_dynamic_children():
-        module = module_creator(descriptor)
+        # do not rebind the module if it's already bound to a user.
+        if descriptor.scope_ids.user_id and user_id == descriptor.scope_ids.user_id:
+            module = descriptor
+        else:
+            module = module_creator(descriptor)
```

Comment:
So `module_creator` now defaults to `None` - how can you be sure not to call a `None` method here? I realize that `None` could have been passed-in before, but it seems even more error-prone to have a parameter defaulting to `None` that's potentially called as a method. There's a hidden assumption here that it won't happen - the assumption should be stated instead.